src/core/server/integration_tests/saved_objects/migrations/zdt_2/sor_higher_version_docs.test.ts

```diff
@@ -69,58 +61,34 @@ describe('Higher version doc conversion', () => {
     });
   };
 
-  const createBaseline = async () => {
-    const testTypeV1 = getTestType({ includeVersion2: false });
-    const testTypeV2 = getTestType({ includeVersion2: true });
+  beforeAll(async () => {
+    await modelVersionTestBed.startES();
 
-    const {
-      runMigrations,
-      savedObjectsRepository: savedObjectsRepositoryV1,
-      client,
-    } = await getKibanaMigratorTestKit({
-      ...getBaseMigratorParams(),
+    const testkit = await modelVersionTestBed.prepareTestKit({
       logFilePath,
-      types: [testTypeV1],
+      savedObjectDefinitions: [
+        {
+          definition: getTestType(),
+          modelVersionBefore: 1,
+          modelVersionAfter: 2,
+        },
+      ],
+      objectsToCreateBetween: range(5).map<SavedObjectsBulkCreateObject>((number) => ({
+        id: `doc-${number}`,
+        type: 'test-type',
+        attributes: {
+          text: `a_${number}`,
+          bool: true,
+        },
+      })),
     });
-    await runMigrations();
-
-    const sampleAObjs = range(5).map<SavedObjectsBulkCreateObject>((number) => ({
-      id: `doc-${number}`,
-      type: 'test-type',
-      attributes: {
-        text: `a_${number}`,
-        bool: true,
-      },
-    }));
-
-    await savedObjectsRepositoryV1.bulkCreate(sampleAObjs, { refresh: 'wait_for' });
-
-    const { runMigrations: runMigrationsAgain, savedObjectsRepository: savedObjectsRepositoryV2 } =
-      await getKibanaMigratorTestKit({
-        ...getBaseMigratorParams(),
-        logFilePath,
-        types: [testTypeV2],
-      });
-    await runMigrationsAgain();
-
-    // returns the repository for v1
-    return { savedObjectsRepositoryV1, savedObjectsRepositoryV2, client };
-  };
-
-  beforeAll(async () => {
-    await fs.unlink(logFilePath).catch(() => {});
-    esServer = await startElasticsearch();
-
-    const { savedObjectsRepositoryV1: sorV1, savedObjectsRepositoryV2: sorV2 } =
-      await createBaseline();
 
-    repositoryV1 = sorV1;
-    repositoryV2 = sorV2;
+    repositoryV1 = testkit.repositoryBefore;
+    repositoryV2 = testkit.repositoryAfter;
   });
 
   afterAll(async () => {
-    await esServer?.stop();
-    await delay(10);
+    await modelVersionTestBed.stopES();
   });
 
```

Comment:
Perhaps I missed something, but are we alright not doing any ES indices cleanup between tests?

I checked the `prepareModelVersionTestKit` and we don't seem to be doing any cleanup there.

The different `runMigrations()` calls inside that method are supposedly updating SO documents to the latest versions, so when the "after" migrator has run once, I would expect all documents to remain in "after" version for subsequent tests?

Reply:
> Perhaps I missed something, but are we alright not doing any ES indices cleanup between tests

You're right. I had this as a todo and then the test I adapted didn't need to run the migration again between each test, so I did not expose an API to delete the indices.

I will add it 👍

Reply:
Added in https://github.com/elastic/kibana/pull/167501/commits/bebdf662f59cd32b715c58c49df2b8341984a176